Source/Core/Common/FileUtil.cpp

```diff
@@ -486,7 +486,7 @@ FSTEntry ScanDirectoryTree(std::string directory, bool recursive)
       }
       else if (cur_depth < prev_depth)
       {
-        while (dir_fsts.size() - 1 != cur_depth)
+        while (static_cast<int>(dir_fsts.size()) - 1 != cur_depth)
```

Comment:
`cur_depth` and `prev_depth`, despite being integers, can never be negative, so the better solution here would be to change the comparison to `dir_fsts.size() != static_cast<size_t>(cur_depth + 1)`.